cmd/juju/application/addrelation.go

```diff
@@ -34,28 +44,26 @@ func (c *addRelationCommand) Info() *cmd.Info {
 }
 
 func (c *addRelationCommand) Init(args []string) error {
+	// Must have only 2 arguments
 	if len(args) != 2 {
 		return errors.Errorf("a relation must involve two applications")
 	}
+	// None of the arguments can be empty.
+	if args[0] == "" || args[1] == "" {
+		return errors.Errorf("a relation must involve two applications")
```

Comment:
This error message isn't great. It's not that a value wasn't specified, but it was *empty*. You really have to be trying to do that. I feel it would be more helpful to validate according to the names package, i.e. check that each arg matches

```
fmt.Sprintf("%s(?::%s)?", names.ApplicationSnippet, names.RelationSnippet)
```